### PR TITLE
Storage container CRD operations added

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
Added create, read and delete functionality for storage container resource
Also added examples for volume in main.tf

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested using main.tf, storage container was created and destroyed successfully

### CODE REVIEW

File: powerstore/resource_storagecontainer.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	storageContainerCreateResponse, err := r.p.client.PStoreClient.CreateStorageContainer(context.Background(), storageContainerCreate)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating Storage Container",
+			"Could not create Storage Container, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	// Get Volume Details using ID retrieved above
```
Comment (mkmayank-dell) [inline powerstore/resource_storagecontainer.go]: comment typo
its for storage container and not volume

File: powerstore/resource_storagecontainer.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+func (r resourceStorageContainer) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+
+	var state models.StorageContainer
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get volume details from API and then update what is in state from what the API returns
```
Comment (mkmayank-dell) [inline powerstore/resource_storagecontainer.go]: same typo for storage container instead of volume

File: powerstore/resource_storagecontainer.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	storageContainerID := state.ID.Value
+	volResponse, err := r.p.client.PStoreClient.GetStorageContainer(context.Background(), storageContainerID)
```
Comment (mkmayank-dell) [inline powerstore/resource_storagecontainer.go]: variable name should be response or related to storage container and not volume

File: powerstore/resource_storagecontainer.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+	log.Printf("Started with Delete")
+
+	var state models.StorageContainer
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get vg ID from state
```
Comment (mkmayank-dell) [inline powerstore/resource_storagecontainer.go]: same comment typo

File: powerstore/resource_storagecontainer.go [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+	storageContainerID := state.ID.Value
+
+	// Delete volume by calling API
```
Comment (mkmayank-dell) [inline powerstore/resource_storagecontainer.go]: comment typo

File: powerstore/resource_storagecontainer.go [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+				MarkdownDescription: "Name for the storage container. This should be unique across all storage containers in the cluster.",
+			},
+			"quota": {
+				Type:                types.Int64Type,
+				Optional:            true,
+				Computed:            true,
+				Description:         "The total number of bytes that can be provisioned/reserved against this storage container.",
+				MarkdownDescription: "The total number of bytes that can be provisioned/reserved against this storage container. A value of 0 means there is no limit. ",
+			},
+			"storage_protocol": {
```
Comment (mkmayank-dell) [inline powerstore/resource_storagecontainer.go:38]: oneOfStringtValidator validator might be used

**Review by mkmayank-dell — APPROVED** [review]

File: powerstore/resource_storagecontainer.go [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting storage container",
+			"Could not delete storageContainerID "+storageContainerID+": "+err.Error(),
+		)
+		return
+	}
+
+	// Remove resource from state
+	resp.State.RemoveResource(ctx)
```
Comment (shenda1) [inline powerstore/resource_storagecontainer.go]: Please check if this statement is required

File: docs/resources/storagecontainer.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "powerstore_storagecontainer Resource - powerstore"
+subcategory: ""
+description: |-
+
```
Comment (shenda1) [inline docs/resources/storagecontainer.md:6]: Add example as well

File: powerstore/resource_storagecontainer.go (hunk 7 above)
Comment (Sakshi-dell) [reply powerstore/resource_storagecontainer.go]: removed

**Review by shenda1 — APPROVED** [review]

**Pull request merged**